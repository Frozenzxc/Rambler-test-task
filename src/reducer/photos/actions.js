const ActionType = {
  LOAD_PHOTOS: `LOAD_PHOTOS`,
};

const ActionCreator = {
  loadPhotos: (photos) => {
    return {
      type: ActionType.LOAD_PHOTOS,
      payload: photos.photos.photo,
    };
  },
};

export {ActionType, ActionCreator};
