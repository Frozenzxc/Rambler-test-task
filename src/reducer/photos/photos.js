import {ActionCreator, ActionType} from "./actions";
import {extend, parsePhoto} from "../../util";

const initialState = {
  photos: [],
  isLoading: true,
};

const Operation = {
  loadPhotos: () => (dispatch, getState, api) => {
    return api.get(`/?method=flickr.interestingness.getList&api_key=b59f451ed922b064e74caf00fa3df0c4&extras=description%2C+license%2C+date_upload%2C+date_taken%2C+owner_name%2C+icon_server%2C+original_format%2C+last_update%2C+geo%2C+tags%2C+machine_tags%2C+o_dims%2C+views%2C+media%2C+path_alias%2C+url_n&per_page=20&page=1&format=json&nojsoncallback=1`)
            .then((response) => {
              dispatch(ActionCreator.loadPhotos(response.data));
            });
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.LOAD_PHOTOS:
      let parsedPhotos = action.payload.map((photo) => parsePhoto(photo));
      return extend(state, {
        isLoading: false,
        photos: parsedPhotos,
      });
  }

  return state;
};

export {reducer, Operation, ActionType, ActionCreator};
