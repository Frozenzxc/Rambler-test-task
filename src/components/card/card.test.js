import React from 'react';
import renderer from 'react-test-renderer';
import Card from "./card";
import photos from "../../test-mocks/test-photos";

const photo = photos[0];

it(`Card is rendered correctly`, () => {
  const tree = renderer.create(
      <Card
        photo={photo}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
