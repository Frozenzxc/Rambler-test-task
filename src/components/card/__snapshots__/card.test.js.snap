// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card is rendered correctly 1`] = `
<li
  className="gallery__item"
>
  <div
    className="gallery__item-header"
  >
    <p
      className="gallery__item-author"
    >
      Dylan Toh
    </p>
    <img
      className="gallery__item-avatar"
      src="http://farm4.staticflickr.com/3171/buddyicons/9619621@N02.jpg"
    />
    <p
      className="gallery__place"
    >
      Bales Beach
    </p>
    <p
      className="gallery__date"
    >
      in a day
    </p>
  </div>
  <img
    className="gallery__item-photo"
    src="https://live.staticflickr.com/65535/50410942841_f23992557b_n.jpg"
  />
  <p
    className="gallery__description"
  >
    During our last trip to KI, I took a walk along the spectacular Bale's Beach. Pristine sands and not a soul around as well as dramatic approaching cloud - all made for a great morning.
  </p>
  <p
    className="gallery__likes"
  >
    <span
      className="gallery__likes-icon"
      onClick={[Function]}
    >
      heart
    </span>
    22869
  </p>
</li>
`;
